Source/cmMakefileExecutableTargetGenerator.cxx

```diff
@@ -613,6 +630,11 @@ void cmMakefileExecutableTargetGenerator::WriteExecutableRule(bool relink)
     this->LocalGenerator->SetLinkScriptShell(false);
   }
 
+  if (this->Makefile->IsOn("CMAKE_EXPORT_LINK_COMMANDS")) {
```

Comment:
Duplicated code: https://github.com/Software-Analysis-Team/CMake/blob/9dcad0c51f680b08f53f75f260934cc53a03b59c/Source/cmMakefileExecutableTargetGenerator.cxx#L263-L266 https://github.com/Software-Analysis-Team/CMake/blob/9dcad0c51f680b08f53f75f260934cc53a03b59c/Source/cmMakefileLibraryTargetGenerator.cxx#L917-L920 https://github.com/Software-Analysis-Team/CMake/blob/9dcad0c51f680b08f53f75f260934cc53a03b59c/Source/cmMakefileLibraryTargetGenerator.cxx#L402-L405
Maybe move it to a separate method?